tests/test_enterprise/management/test_backfill_ecu_table_user_foreign_key.py

```diff
@@ -0,0 +1,129 @@
+        call_command(self.command)
+        assert EnterpriseCustomerUser.objects.first().user_fk is EnterpriseCustomerUser.objects.first().user_id
+
+    @patch('logging.Logger.info')
+    @patch('enterprise.management.commands.backfill_ecu_table_user_foreign_key.sleep')
+    def test_runs_in_batches(self, mock_sleep, mock_log):
+        ecus = EnterpriseCustomerUser.objects.all()
+        for ecu in ecus:
+            ecu.user_fk = None
+        EnterpriseCustomerUser.objects.all().bulk_update(ecus, ['user_fk'])
+
+        call_command(self.command, batch_limit=3)
+        assert mock_sleep.call_count == 4
+        mock_log.assert_any_call('Processed 3 records.')
+        mock_log.assert_any_call('Processed 6 records.')
+        mock_log.assert_any_call('Processed 9 records.')
+        mock_log.assert_any_call('Processed 12 records.')
+
+    def test_skips_rows_that_already_have_user_fk(self):
+        ecu = EnterpriseCustomerUser.objects.first()
+        ecu.user_fk = 9999
+        # use bulk_update to prevent save() method from setting user_fk to user_id
+        EnterpriseCustomerUser.objects.all().bulk_update([ecu], ['user_fk'])
+        call_command(self.command)
+        assert EnterpriseCustomerUser.objects.first().user_fk == 9999
+
+    @patch('logging.Logger.warning')
+    def test_retry_5_times_on_failure(self, mock_log):
+        ecus = EnterpriseCustomerUser.objects.all()
+        for ecu in ecus:
+            ecu.user_fk = None
+        EnterpriseCustomerUser.objects.all().bulk_update(ecus, ['user_fk'])
+
+        with patch(
+            ("enterprise.management.commands.backfill_ecu_table_user_foreign_key." +
+                "EnterpriseCustomerUser.objects.bulk_update"),
+            side_effect=DatabaseError(EXCEPTION)
+        ):
+            with self.assertRaises(Exception):
+                call_command(self.command, max_retries=2)
+            mock_log.assert_any_call('Attempt 1/2 failed: DUMMY_TRACE_BACK. Retrying in 2s.')
+            mock_log.assert_any_call('Attempt 2/2 failed: DUMMY_TRACE_BACK. Retrying in 4s.')
+
+    @patch("enterprise.management.commands.backfill_ecu_table_user_foreign_key._fetch_and_update_in_batches")
+    def test_doesnt_load_all_rows_into_memory(self, mock_fetch_and_update):
+        """
+        As long as the queryset is not evaluated, database rows won't be fetched into memory.
+        Since the `iterator` method is used in the `_fetch_and_update_in_batches` function, the rows
+        are fetched in chunks.
+        This test ensures that the QuerySet is not evaluated before being passed to `_fetch_and_update_in_batches`.
+        """
+        call_command(self.command, batch_limit=3)
+
+        _, kwargs = mock_fetch_and_update.call_args
+        queryset = kwargs.get("queryset")
+
+        # Check that QuerySet was NOT evaluated before `_fetch_and_update_in_batches`
+        assert queryset._result_cache is None, "QuerySet was evaluated too early!"  # pylint: disable=protected-access
```

Comment:
The reason this test works is that the unit tests use a sqlite3 database, and not mysql.

Reply:
Then I'm removing the test. Otherwise it will only produce false positives.